ultipro/operations.py

```diff
@@ -0,0 +1,363 @@
+        employeePhoneList = []
+        if pageNumber != "all":
+            response_dict = self.__findEmployeePhone(pageNumber, pageSize)
+            return response_dict
+        else:
+            pageNumber = 1
+            response_dict = self.__findEmployeePhone(pageNumber, pageSize)
+            pageTotal = response_dict["s:Envelope"]["s:Body"][
+                "FindPhoneInformationsResponse"
+            ]["FindPhoneInformationsResult"]["b:OperationResult"]["b:PagingInfo"][
+                "b:PageTotal"
+            ]
+            while pageNumber <= int(pageTotal):
+                pageNumber += 1
+                employeePhoneList += response_dict["s:Envelope"]["s:Body"][
+                    "FindPhoneInformationsResponse"
+                ]["FindPhoneInformationsResult"]["b:Results"][
+                    "b:EmployeePhoneInformation"
+                ]
+                response_dict = self.__findEmployeePhone(pageNumber, pageSize)
+            return employeePhoneList
+
+    def __findEmployeePerson(self, pageNumber, pageSize):
+        # This function will perform a generic search to return all people in the company
+        headers = {"Content-Type": "application/soap+xml"}
+        body = f"""<s:Envelope xmlns:a="http://www.w3.org/2005/08/addressing" xmlns:s="http://www.w3.org/2003/05/soap-envelope">
+        <s:Header>
+            <a:Action s:mustUnderstand="1">http://www.ultipro.com/services/employeeperson/IEmployeePerson/FindPeople</a:Action>
+            <UltiProToken xmlns="http://www.ultimatesoftware.com/foundation/authentication/ultiprotoken">{self.ultiProToken}</UltiProToken>
+            <ClientAccessKey xmlns="http://www.ultimatesoftware.com/foundation/authentication/clientaccesskey">{self.clientApiKey}</ClientAccessKey>
+        </s:Header>
+        <s:Body>
+            <FindPeople xmlns="http://www.ultipro.com/services/employeeperson">
+            <query xmlns:b="http://www.ultipro.com/contracts" xmlns:i="http://www.w3.org/2001/XMLSchema-instance">
+                <b:CompanyCode> = {self.companyId}</b:CompanyCode>
+                <b:PageNumber> {pageNumber}</b:PageNumber>
+                <b:PageSize> {pageSize}</b:PageSize>
+            </query>
+            </FindPeople>
+        </s:Body>
+        </s:Envelope>"""
+        try:
+            response = requests.post(
+                "https://service5.ultipro.com/services/EmployeePerson",
+                data=body,
+                headers=headers,
+            )
+            if response.ok:
+                response_dict = xmltodict.parse(response.text)
+                return response_dict
+            else:
+                raise ConnectorError(response.text)
+        except requests.exceptions.ConnectionError as e:
+            logger.exception("{}".format(e))
+            raise ConnectorError(
+                "{}".format(
+                    "The request timed out while trying to connect to the remote server"
+                )
+            )
```

Comment:
you can optimise the code. Please write a common method to handle the api request.